src/pages/youthProfiles/create/inputs/BirthDateInput.tsx

```diff
@@ -0,0 +1,103 @@
+import React, { useState, useEffect } from 'react';
+import { TextInput } from 'hds-react';
+import { Box } from '@material-ui/core';
+import { makeStyles } from '@material-ui/styles';
+import { useField } from 'react-final-form';
+import { useTranslate } from 'react-admin';
+
+const useStyles = makeStyles({
+  birthDate: { width: 91, marginRight: '1rem' },
+  label: {
+    fontSize: 'var(--hds-text-sm)',
+    lineHeight: 'var(--hds-text-sm-line-height)',
+    fontFamily: 'var(--hds-theme-primary-font)',
+    fontWeight: 'bold',
+    margin: 0,
+  },
+  error: { color: '#c4123e' },
+  errorHelper: {
+    color: '#c4123e',
+    margin: '2px 0',
+    fontSize: '0.875rem',
+  },
+  container: { minWidth: 305 },
+});
+
+type Props = {
+  inputName: string;
+  label: string;
+};
+
+type BirthDate = {
+  day?: string;
+  month?: string;
+  year?: string;
+};
+
+const BirthDateInput = ({ inputName, label }: Props) => {
+  const classes = useStyles();
+  const [birthDate, setBirthDate] = useState<BirthDate>({
+    day: '',
+    month: '',
+    year: '',
+  });
+
+  const {
+    input: { onChange },
+    meta: { touched, error },
+  } = useField(inputName);
+
+  useEffect(() => {
+    onChange({
+      target: {
+        value: `${birthDate.year}-${birthDate.month}-${birthDate.day}`,
+      },
+    });
+  }, [birthDate, onChange]);
+
+  // e type is set to any for now. Event type returned from hds
+  // is set to ChangeEvent<Element> which doesn't contain
+  // target.value
+  /* eslint-disable-next-line  @typescript-eslint/no-explicit-any */
+  const handleChange = (e: any) => {
+    setBirthDate({ ...birthDate, [e.target.id]: e.target.value });
```

Comment:
You could consider using the function setter here to get reliable access to the current `birthDate`. In this context the difference likely won't matter, but you may get some unexpected behaviour if users update the different birthday components in quick succession.

```suggestion
    setBirthDate(previousBirthDate => ({ ...previousBirthDate, [e.target.id]: e.target.value }));
```

Reply:
Added `prevState`.